init.c

```diff
@@ -281,28 +281,24 @@ static int complete_all_nm_tags(const char *pt)
  */
 static int execute_commands(struct ListHead *p)
 {
-  struct Buffer err, token;
+  int rc = 0;
+  struct Buffer *err = mutt_buffer_alloc(256);
+  struct Buffer *token = mutt_buffer_alloc(256);
 
-  mutt_buffer_init(&err);
-  err.dsize = 256;
-  err.data = mutt_mem_malloc(err.dsize);
-  mutt_buffer_init(&token);
   struct ListNode *np = NULL;
   STAILQ_FOREACH(np, p, entries)
   {
-    if (mutt_parse_rc_line(np->data, &token, &err) == MUTT_CMD_ERROR)
+    if (mutt_parse_rc_line(np->data, token, err) == MUTT_CMD_ERROR)
     {
-      mutt_error(_("Error in command line: %s"), err.data);
-      FREE(&token.data);
-      FREE(&err.data);
-
-      return -1;
+      mutt_error(_("Error in command line: %s"), mutt_b2s(err));
+      rc = 1;
+      break;
```

Comment:
[1] Everything from [0], but this changes the error return code from this function as well.